### DISCUSSION TITLE
Using Maester for checking Entra user attributes

### DISCUSSION THREAD

**Discussion by kentishh** [maester365/maester#500] [General]
Hey guys! I wanted to extend Maester usage for checking certain attributes are populated against Entra users, such as Manager, job title etc.

Would Maester be a good choice for this? Could Maester in theory output the report, then tell us which people are missing certain attributes we define?

A code snippet would also be really useful if anyone has any idea if this would work or not, a lot of my trial and error hasn't quite worked as expected. Cheers

**Comment by soulemike** [comment]
Yep, there are a few ways this could be handled. Each could be one or more hygiene tests.

* Distribution of value existing # of objects with value/# total objects
* List specific objects missing a value
* List specific objects with an incorrect/undefined value

**Comment by DevClate** [comment]
> Hey guys! I wanted to extend Maester usage for checking certain attributes are populated against Entra users, such as Manager, job title etc.
> 
> Would Maester be a good choice for this? Could Maester in theory output the report, then tell us which people are missing certain attributes we define?
> 
> A code snippet would also be really useful if anyone has any idea if this would work or not, a lot of my trial and error hasn't quite worked as expected. Cheers

Hi! Working on this now, I had created a module ( https://github.com/DevClate/365AutomatedCheck )that does this and will even allow you to correct invalid fields on the fly, but for now converting all the tests to Maester formatting. I'm starting with these tests and will add more and more formatting options too.
